packages/server/src/data/db/db-operations/post-ops.js

```diff
@@ -134,6 +165,66 @@ const downvotePost = async (postId, userId) => {
   });
 };
 
+const upvoteComment = async (postId, commentId, userId) => {
+  const updateSpec = {
+    $push: { "comments.$[comment].upvoters": userId },
+    $pull: { "comments.$[comment].downvoters": userId },
+  };
+
+  const filter = {
+    _id: postId,
+  };
+
+  const options = {
+    new: true,
+    projection: createProjectionObject(userId, true).$project,
+    arrayFilters: [
+      {
+        "comment._id": commentId,
+        "comment.upvoters": { $nin: [userId] },
+      },
+    ],
+  };
+
+  const post = await Post.findOneAndUpdate(filter, updateSpec, options);
+  const comment = await post.comments.find(
+    (comment) => comment._id == commentId,
+  );
+  updateUserReputation(comment.userId, 1);
+
+  return comment;
+};
+
+const downvoteComment = async (postId, commentId, userId) => {
+  const updateSpec = {
+    $push: { "comments.$[comment].downvoters": userId },
+    $pull: { "comments.$[comment].upvoters": userId },
+  };
+
+  const filter = {
+    _id: postId,
+  };
+
+  const options = {
+    new: true,
+    projection: createProjectionObject(userId, true).$project,
+    arrayFilters: [
+      {
+        "comment._id": commentId,
+        "comment.downvoters": { $nin: [userId] },
+      },
+    ],
+  };
+
+  const post = await Post.findOneAndUpdate(filter, updateSpec, options);
+  const comment = await post.comments.find(
+    (comment) => comment._id == commentId,
+  );
+  updateUserReputation(comment.userId, -1);
```

Comment:
Hmm maybe we should just return `new: false` to check how much to increment/decrement by. `new: false` would return the post before the update. We can see if they were already in upvoters and if they are then decrement by 2, otherwise increment by 1.